src/gui/basegraphgl.cpp

```diff
@@ -381,6 +391,34 @@ void BaseGraphGL::mouseReleaseEvent(QMouseEvent *e)
     }
 }
 
+
+void BaseGraphGL::updateFullInspector()
```

Comment:
as mentioned before... the `BaseGraphGL` should be kept independent of the inspectors.

That is, this class should not be changing anything in the `FullInspector` because it'd introduce a flaw in the architecture. Instead, it could just emit a signal to let the listeners know when something was selected. Then, the listeners should do the job of making the FullInspector aware that its contents should be updated.

Note that `BaseGraphGL` is instantiated by the `GraphWidget` class, which is instantiated by the `GraphDesigner` class. The `GraphDesigner` in turn is instantiated by the `GraphDesignerPage` class, which also instantiates the `FullInspector` , i.e.,

`GraphDesignerPage` -> `GraphDesigner` -> `GraphWidget` -> `BaseGraphGL`

`GraphDesignerPage` -> `FullInspector`

Thus,
- the `FullInspector` could have a public slot like `void slotSelectedNode(const Node& node);` with the logic to update itself with the information provided by the selected node
- the `BaseGraphGL` could emit a `nodeSelectionChanged(node);`
- the `GraphDesignerPage` could make the connection between them